src/Aspire.Hosting.Kubernetes/KubernetesResourceContext.cs

```diff
@@ -0,0 +1,368 @@
+            {
+                throw new InvalidOperationException("Volume source and target must be set");
+            }
+
+            if (volume.Type == ContainerMountType.BindMount)
+            {
+                throw new InvalidOperationException("Bind mounts are not supported by the Kubernetes publisher");
+            }
+
+            var newVolume = new VolumeMountV1
+            {
+                Name = volume.Source,
+                ReadOnly = volume.IsReadOnly,
+                MountPath = volume.Target,
+            };
+
+            Volumes.Add(newVolume);
+        }
+    }
+
+    private async Task ProcessArgumentsAsync(CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<CommandLineArgsCallbackAnnotation>(out var commandLineArgsCallbackAnnotations))
+        {
+            var context = new CommandLineArgsCallbackContext([], cancellationToken: cancellationToken);
+
+            foreach (var c in commandLineArgsCallbackAnnotations)
+            {
+                await c.Callback(context).ConfigureAwait(false);
+            }
+
+            foreach (var arg in context.Args)
+            {
+                var value = await ProcessValueAsync(arg).ConfigureAwait(false);
+
+                if (value is not string str)
+                {
+                    throw new NotSupportedException("Command line args must be strings");
+                }
+
+                Commands.Add(new(str));
+            }
+        }
+    }
+
+    private async Task ProcessEnvironmentAsync(DistributedApplicationExecutionContext executionContext, CancellationToken cancellationToken)
+    {
+        if (resource.TryGetAnnotationsOfType<EnvironmentCallbackAnnotation>(out var environmentCallbacks))
+        {
+            var context = new EnvironmentCallbackContext(executionContext, cancellationToken: cancellationToken);
+
+            foreach (var c in environmentCallbacks)
+            {
+                await c.Callback(context).ConfigureAwait(false);
+            }
+
+            var processedKeys = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var kv in context.EnvironmentVariables)
+            {
+                var value = await ProcessValueAsync(kv.Value).ConfigureAwait(false);
+                var key = kv.Key.ToManifestFriendlyResourceName();
+                var stringValue = value.ToString() ?? string.Empty;
+
+                if (processedKeys.Contains(key))
+                {
+                    continue;
+                }
+
+                // Move connection strings to secrets
+                if (key.IsConnectionString())
+                {
+                    var expression = key.ToHelmSecretExpression(resource.Name);
+                    Secrets[key] = new HelmExpressionValue(expression, stringValue);
+                    continue;
+                }
+
+                if (stringValue.IsHelmSecretExpression())
```

Comment:
How does this happen?

Reply:
this occurs after helm expressions have been constructed in variables,
So for instance, if we take the Parameter param1 from the 'myapp' test app, it has a kubernetes secret generated, which includes a helm expression of '{{ .Values.secrets.myapp.param1 }}' as its value

Then the final values file for the helm chart with deploy time ability to set values, can supply that value

Reply:
I don't fully understand how. This is processing the environment variables of a resource, right? The value can't be a help expression at this stage unless the user or some process directly set a helm expression in env. What am I missing?

Reply:
Ah I seem this runs on the result of the `ProcessValue` call? Why not use a strongly typed object to return the helm expression instead of converting to a string first?

Reply:
Its because a little above, ProcessValueAsync is called, which processes, and transforms the end `stringValue`, so when you get to that line, its already gone through AllocateParameter etc

Reply:
Thats a good shout, as we already construct one (HelmExpressionValue) which is just a struct with the Expression, and Value